pkg/shell/cockpit-accounts.js

```diff
@@ -465,7 +474,90 @@ PageAccountsCreate.prototype = {
             });
 
         $("#accounts-create-dialog").dialog("wait", promise);
+    },
+
+    is_valid_char_username: function(c) {
+        return (c >= 'a' && c <= 'z') ||
+               (c >= 'A' && c <= 'Z') ||
+               (c >= '0' && c <= '9');
+    },
+
+    check_username: function() {
+        var dfd = $.Deferred();
+        var username = $('#accounts-create-user-name').val();
+
+        for (var i = 0; i < username.length; i++) {
+            if (! this.is_valid_char_username(username[i])) {
+                dfd.reject(new Error(_("User name should consist of letters without diacritics")));
+                return dfd.promise();
+            }
+        }
+
+        cockpit.spawn(['/usr/bin/grep', '-q', "^" + username + ":", '/etc/passwd'])
```

Comment:
Why can't we do this from our list in memory? Since we load the entire /etc/passwd could we track that and check against it?